src/hotspot/cpu/aarch64/codeBuffer_aarch64.cpp

```diff
@@ -26,6 +26,52 @@
 #include "asm/codeBuffer.inline.hpp"
 #include "asm/macroAssembler.hpp"
 
+void CodeBuffer::share_trampoline_for(address dest, int caller_offset) {
+  if (_shared_trampoline_requests == nullptr) {
+    _shared_trampoline_requests = new SharedTrampolineRequests(4, 1024);
```

Comment:
The meaning of `4` and `1024` is not clear. It requires to go to `ResizeableResourceHashtable` definition. Comments or named constants would be useful.
`ResizeableResourceHashtable` uses open hashing. The first parameter is a number of buckets. The second parameter is the maximum number of buckets. `maybe_grow` is to rebalance a hash table to reduce lengths of linked lists.
I think you don't need `1024` at all. A reasonable number of buckets should be enough.